models/models.go

```diff
@@ -8,28 +8,27 @@ import (
 type FetchMeta struct {
 	ID uint `gorm:"primary_key"`
 
-	FileName  string
+	FileName  string `gorm:"type:varchar(255)"`
 	Timestamp time.Time
 }
 
 // Root is root struct
 type Root struct {
-	ID uint `gorm:"primary_key"`
-	//  Timestamp   time.Time
-	Family      string
-	OSVersion   string
+	ID          uint   `gorm:"primary_key"`
+	Family      string `gorm:"type:varchar(255)"`
+	OSVersion   string `gorm:"type:varchar(255)"`
 	Definitions []Definition
 	Timestamp   time.Time
 }
 
 // Definition : >definitions>definition
 type Definition struct {
-	ID     uint `gorm:"primary_key"`
-	RootID uint `json:"-" xml:"-"`
+	ID     uint `gorm:"primary_key" json:"-"`
+	RootID uint `gorm:"index:idx_definition_root_id" json:"-" xml:"-"`
 
-	DefinitionID  string
+	DefinitionID  string `gorm:"type:varchar(255)"`
 	Title         string `gorm:"type:text"`
```

Comment:
Is this text struct tag required?

Reply:
Don't you want all columns to be allocated with longtext?